frontend/packages/console-app/src/actions/creators/hpa-factory.ts

```diff
@@ -93,7 +88,9 @@ type DeploymentActionExtraResources = {
 };
 
 export const useHPAActions = (kindObj: K8sKind, resource: K8sResourceKind) => {
-  const [namespace] = useActiveNamespace();
+  const {
+    metadata: { namespace },
+  } = resource;
```

Comment:
Destructuring is more error-prone than using optional chaining:
```suggestion
  const namespace = resource?.metadata?.namespace;
```

We probably also need to handle the case where `namespace` is undefined. We didn't have this issue with the `useActiveNamespace` hook because it always returns something. In this case, it's possible that the request for the resource being passed to this hook has not returned yet.